test/e2e/framework/pod/wait.go

```diff
@@ -153,41 +155,31 @@ func WaitForPodsRunningReady(ctx context.Context, c clientset.Interface, ns stri
 			Pods:                   podList.Items,
 		}, nil
 	})).WithTimeout(timeout).Should(framework.MakeMatcher(func(s *state) (func() string, error) {
-		replicas, replicaOk := int32(0), int32(0)
-		for _, rc := range s.ReplicationControllers {
-			replicas += *rc.Spec.Replicas
-			replicaOk += rc.Status.ReadyReplicas
-		}
-		for _, rs := range s.ReplicaSets {
-			replicas += *rs.Spec.Replicas
-			replicaOk += rs.Status.ReadyReplicas
-		}
 
-		nOk := int32(0)
-		notReady = int32(0)
-		failedPods := []v1.Pod{}
-		otherPods := []v1.Pod{}
-		succeededPods := []string{}
+		nOk = 0
+		badPods = []v1.Pod{}
+		otherPods = []v1.Pod{}
+		succeededPods = []string{}
 		for _, pod := range s.Pods {
 			res, err := testutils.PodRunningReady(&pod)
 			switch {
 			case res && err == nil:
 				nOk++
 			case pod.Status.Phase == v1.PodSucceeded:
-				// it doesn't make sense to wait for this pod
+				// ignore succeeded pods
 				succeededPods = append(succeededPods, pod.Name)
 			case pod.Status.Phase == v1.PodFailed:
 				// ignore failed pods that are controlled by some controller
+				// TODO either document why failures with controllers are allowed while
```

Comment:
I have absolutely no idea why there exists this special carve-out for Failed Pods with vs without a Controller. Does anyone know any reason to not treat all Failed Pods as failures? This seemed like the sort of thing that was added for a specific reason, and felt out of scope to determine whether or not was appropriate to change.

Reply:
Let's not add TODOs without a clear path towards resolving them.

I suspect the motivation might have been that failed pods owned by a controller will get deleted and recreated, but I am not sure.

Reply:
Will remove! I've always had a strong bias towards actively calling out "no one who has recently looked at this code is completely sure why this is written this way", since it makes immediately clearer to readers the distiction between "core logic of the function" and "logic which hasn't been changed only because people are scared to touch it". It would have saved me trouble to have this explicitily called out when reading this code!

Of course, I'm more than happy to defer to you for such style concerns.

Reply:
I don't think we as maintainers later on add "what the heck is this code doing" comments to existing code just because we touch it and don't understand what the original author meant with it. Unless something is very obviously broken, the code defines the behavior of a method and shouldn't be changed without considering all implications.

It is not unlikely that you will run into situations like that. Happens to me all the time... :smile: But then try to clarify with the original author or people who might know via some other channels (Slack or in a review comment on your own PR) instead of adding TODO comments.

Reply:
Haha I appreciate that response, thank you!